### PR TITLE
Remove lodash-es 

### LINKED ISSUE #171: Remove lodash dependency
Reference: https://github.com/reakit/reakit/pull/170#issuecomment-410509334

I think that's a good idea to remove the dependency on lodash and create util modules instead. We're currently using `kebabCase`, `uniq`, `uniqueId` and `omit`. We should create **one file per function** and test them.

An initial work was done by @Andarist here:
https://github.com/reakit/reakit/blob/f688d223bd8888691577b3a19ead02c475b75dab/src/utils/lodash-like.js#L1-L25

### PR DESCRIPTION
**Pull request by Thomazella**
fixed #171 

Removes lodash-es and replaces its functions with utils. @andarist kickstarted the code for them. I added tests, and changed things.

This PR also refactors benchmark/index.js. Now we can use `yarn benchmark step` and it'll run benchmarks including 'step' in the name. Also improved benchmark output to be nicer. Old system using 'only' and 'skip' was removed.

### CODE REVIEW

File: src/utils/__tests__/kebabCase-test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+test("converts common cases", () => {
+  expect(kebab("foobar")).toEqual("foobar");
+  expect(kebab("fooBar")).toEqual("foo-bar");
+  expect(kebab("foo_bar")).toEqual("foo-bar");
+  expect(kebab("FooBar")).toEqual("foo-bar");
+  expect(kebab("fooBar")).toEqual("foo-bar");
+  expect(kebab("foo bar")).toEqual("foo-bar");
+  expect(kebab("FOO BAR")).toEqual("foo-bar");
+  expect(kebab("foo-bar")).toEqual("foo-bar");
```
Comment (diegohaz) [inline src/utils/__tests__/kebabCase-test.js]: You can use `.toBe` when you're asserting on literal strings.

File: src/utils/__tests__/kebabCase-test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+});
+
+test("converts edge cases", () => {
+  expect(kebab("____").length).toEqual(0);
+  expect(kebab("----").length).toEqual(0);
+  expect(kebab("     ").length).toEqual(0);
```
Comment (diegohaz) [inline src/utils/__tests__/kebabCase-test.js]: Why not `expect(kebab("     ")).toBe("")`?

File: src/utils/__tests__/omit-test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+test("arg string", () => {
+  const omitted = keys(omit(fixture, "animal"));
+  expect(omitted.includes("animal")).toBeFalsy();
+});
+
+test("arg string[]", () => {
+  const omitted = keys(omit(fixture, ["animal", "age"]));
+  expect(omitted.includes("animal") && omitted.includes("age")).toBeFalsy();
```
Comment (diegohaz) [inline src/utils/__tests__/omit-test.js]: Let's put each assertion on its own line.

File: src/utils/__tests__/kebabCase-test.js (hunk 1 above)
Comment (rthomazel) [reply src/utils/__tests__/kebabCase-test.js]: I'm afraid because `toBe` is kinda strict. I'll change it. :)

File: src/utils/__tests__/uniq-test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import uniq from "../uniq";
+
+test("removes duplicates", () => {
+  expect(uniq([1, 1, 1, 1, 3, 1, 1, 1]).length).toEqual(2);
```
Comment (diegohaz) [inline src/utils/__tests__/uniq-test.js]: I think it's safer to test the actual values:
```
expect(uniq([1, 1, 1, 1, 3, 1, 1, 1])).toEqual([1, 3])
```

Otherwise, `[1, 1]` would pass.

File: src/utils/__tests__/uniqueId-test.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import id from "../uniqueId";
```
Comment (diegohaz) [inline src/utils/__tests__/uniqueId-test.js]: It should be `uniqueId`

File: src/utils/__tests__/uniq-test.js (hunk 4 above)
Comment (rthomazel) [reply src/utils/__tests__/uniq-test.js]: that's  true

File: src/utils/omit.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const omit = (target, undesired = []) => {
+  let result = {};
+  Object.keys(target).forEach(key => {
+    if (!undesired.includes(key))
```
Comment (diegohaz) [inline src/utils/omit.js]: I'm not sure if that's safe to use `includes` here. I'd go for `indexOf`.

And maybe we can use `reduce` instead of `forEach`.

```js
Object.keys(target).reduce((finalObject, key) => ({
  ...finalObject,
  ...(undesired.indexOf(key) === -1 ? { [key]: target[key] } : {})
}), {});
```

File: src/utils/uniqueId.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+let id = -1;
+const uniqueId = prefix => {
+  id += 1;
+  return prefix ? `${prefix}${id}` : id;
+};
+
+export default uniqueId;
```
Comment (diegohaz) [inline src/utils/uniqueId.js:7]: I know that's how lodash does it, but now looking at it I'm wondering if it wouldn't break on server side.

File: styleguide.dev.config.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const omit = require("lodash/omit");
+const omit = require("./src/utils/omit");
```
Comment (diegohaz) [inline styleguide.dev.config.js]: We can use `lodash` here.

File: website/src/utils/filterSections.js [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import kebabCase from "lodash/kebabCase";
+import kebabCase from "../../../src/utils/kebabCase";
```
Comment (diegohaz) [inline website/src/utils/filterSections.js]: We can use `lodash` here.

File: website/src/utils/parseSections.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
-import kebabCase from "lodash/kebabCase";
 import get from "lodash/get";
+import kebabCase from "../../../src/utils/kebabCase";
```
Comment (diegohaz) [inline website/src/utils/parseSections.js]: We can use `lodash` here.

File: src/utils/uniqueId.js (hunk 7 above)
Comment (rthomazel) [reply src/utils/uniqueId.js:7]: What do you mean?

File: src/utils/omit.js (hunk 6 above)
Comment (diegohaz) [reply src/utils/omit.js]: `Array.prototype.includes` works on all modern browsers. It doesn't work on IE 11 (but we're not completely supporting it, so...): https://caniuse.com/#feat=array-includes

Also, [babel doesn't transpile it](https://babeljs.io/en/repl#?babili=false&browsers=ie%206&build=&builtIns=false&spec=false&loose=false&code_lz=NoRgNABATJDMC6A6AlgOwMYBsCuATApgM4AUUAlANxA&debug=true&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=&version=6.26.0&envVersion=1.6.2)

File: src/utils/uniqueId.js (hunk 7 above)
Comment (diegohaz) [reply src/utils/uniqueId.js:7]: References:
- https://github.com/FormidableLabs/victory/issues/664
- https://stackoverflow.com/questions/50231301/prop-mismatch-server-client-how-to-generate-a-unique-id-that-persists
- https://github.com/facebook/react/issues/5867

File: src/utils/uniq.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+const uniq = (arr = []) => {
```
Comment (Andarist) [inline src/utils/uniq.js:1]: is the default here worth anything?

File: src/utils/omit.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+const omit = (target, undesired = []) => {
```
Comment (Andarist) [inline src/utils/omit.js:1]: have u tried to use object rest spread as omit? would save few bytes 😉
